space2stats_api/app/routers/api.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Dict, Any, Literal
+
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel, field_validator
+from shapely.geometry import shape, mapping
+
+from app.utils.h3_utils import generate_h3_ids
+from app.utils.db_utils import get_available_fields, get_summaries
+
+
+router = APIRouter()
+
+class AOI(BaseModel):
+    type: Literal["Polygon"]
+    coordinates: List[List[List[float]]]
+
+    @field_validator("type")
+    def validate_type(cls, v):
+        if v != "Polygon":
+            raise ValueError("aoi must be a polygon")
+        return v
+
+    @field_validator("coordinates")
+    def validate_coordinates(cls, v):
+        try:
+            geom = shape({"type": "Polygon", "coordinates": v})
+            if not geom.is_valid:
+                raise ValueError("invalid coordinates for polygon")
+        except Exception as e:
+            raise ValueError("invalid coordinates for polygon") from e
+        return v
+
+class SummaryRequest(BaseModel):
+    aoi: AOI
+    spatial_join_method: Literal["touches", "centroid", "within"]
+    fields: List[str]
+
+class SummaryResponse(BaseModel):
+    hex_id: str
+    fields: Dict[str, Any]
+
+@router.post("/summary", response_model=List[SummaryResponse])
+def get_summary(request: SummaryRequest):
+    try:
+        geom = shape({"type": "Polygon", "coordinates": request.aoi.coordinates})
+        if not geom.is_valid:
+            raise ValueError("invalid geojson polygon")
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=str(e))
+
+    resolution = 6
+    aoi_geojson = mapping(geom)
```

Comment:
same if we use geojson-pydantic we might not need shapely functions

Reply:
So, we don't need them in `api.py`. I updated the code to use `geojson_pydantic` instead. However, Shapely is still used in the `h3_utils.py` for the spatial join function. We could off-load this logic to Postgres or use another method. I'm not sure if we would want to include those changes in this PR. @vincentsarago @bitner.